src/software/SfM/main_openMVG2ABC.cpp

```diff
@@ -0,0 +1,168 @@
+    OObject topObj(archive, kTop);
+
+    //=================================================================================
+    // Export points
+    //=================================================================================
+
+    std::vector<V3f> positions;
+    positions.reserve(sfm_data.GetLandmarks().size());
+
+    // For all the 3d points in the hash_map
+    for(const auto lm : sfm_data.GetLandmarks()) {
+        const openMVG::Vec3 &pt = lm.second.X;
+        positions.emplace_back(pt[0], pt[1], pt[2]);
+    }
+
+    std::vector<Alembic::Util::uint64_t> ids(positions.size());
+    iota(begin(ids), end(ids), 0);
+
+    OPoints partsOut(topObj, "particleShape1");
+    OPointsSchema &pSchema = partsOut.getSchema();
+    OPointsSchema::Sample psamp( move(V3fArraySample (positions)), move(UInt64ArraySample ( ids )));
+    pSchema.set( psamp );
+
+    //=================================================================================
+    // Export cameras
+    //=================================================================================
+    for(const auto it: sfm_data.GetViews()) {
+        const View * view = it.second.get();
+        if (!sfm_data.IsPoseAndIntrinsicDefined(view))
+          continue;
+
+        // OpenMVG Camera
+        const openMVG::geometry::Pose3 pose = sfm_data.GetPoseOrDie(view);
+        auto iterIntrinsic = sfm_data.GetIntrinsics().find(view->id_intrinsic);
+        openMVG::cameras::Pinhole_Intrinsic *cam = static_cast<openMVG::cameras::Pinhole_Intrinsic*>(iterIntrinsic->second.get());
+        openMVG::Mat34 P = cam->get_projective_equivalent(pose);
+
+        // Extract internal matrix, rotation and translation
+        openMVG::Mat3 R, K;
+        openMVG::Vec3 t;
+        openMVG::KRt_From_P(P, &K, &R, &t);
+        
+        // POSE
+        // Compensate translation with rotation
+        openMVG::Vec3 center = R.transpose() * t;
+        // Build transform matrix
+        Abc::M44d xformMatrix;
+        xformMatrix[0][0]= R(0,0);
+        xformMatrix[0][1]= R(0,1);
+        xformMatrix[0][2]= R(0,2);
+        xformMatrix[1][0]= R(1,0);
+        xformMatrix[1][1]= R(1,1);
+        xformMatrix[1][2]= R(1,2);
+        xformMatrix[2][0]= R(2,0);
+        xformMatrix[2][1]= R(2,1);
+        xformMatrix[2][2]= R(2,2);
+        xformMatrix[3][0]= center(0);
+        xformMatrix[3][1]= center(1);
+        xformMatrix[3][2]= center(2);
+        xformMatrix[3][3]= 1.0;
+
+        // Get correct camera orientation
+        M44d scale;
+        scale[0][0] = -1;
+        scale[1][1] = -1;
+        scale[2][2] = -1;
+
+        M44d scale2;
+        scale2[0][0] = -1;
+        scale2[1][1] = 1;
+        scale2[2][2] = 1;
+        
+        xformMatrix = scale2*xformMatrix*scale;
+
+        XformSample xformsample;
+        xformsample.setMatrix(xformMatrix);
+
+        stringstream ss;
+        ss << stlplus::basename_part(view->s_Img_path); 
```

Comment:
I suppose we can add an extra field in the camera object to keep the image file name but I still have to investigate if it is really possible. It might be a good addition though.